src/v3/src/hooks/usePolling.ts

```diff
@@ -81,17 +92,18 @@ export const usePolling = (
     // one time request
     // the following polling requests will be triggered based on idxTransaction update
     timerRef.current = setTimeout(async () => {
-      // TODO: Revert to use action once this fix is completed OKTA-512706
       let payload: IdxActionParams = {};
-      if (data.autoChallenge !== undefined) {
-        payload.autoChallenge = data.autoChallenge as boolean;
+      const autoChallenge = getAutoChallengeValue(idxTransaction, data);
+      if (autoChallenge !== undefined) {
+        payload.autoChallenge = autoChallenge;
       }
       // POLL_STEPS are not an action, so must treat as such
       if (POLL_STEPS.includes(name)) {
```

Comment:
Left a similar comment on #2720. Extract to utility function and use a `Set` to check for inclusion.